File: src/main/java/com/example/hnefatafl/config/MongoConfig.java
package com.example.hnefatafl.config;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.mongodb.core.MongoTemplate;

@Configuration
public class MongoConfig {

    @Bean
    public MongoClient mongoClient(@Value("${spring.data.mongodb.uri}") String uri) {
        return MongoClients.create(uri);
    }

    @Bean
    public MongoTemplate mongoTemplate(MongoClient mongoClient,
                                       @Value("${spring.data.mongodb.database}") String database) {
        return new MongoTemplate(mongoClient, database);
    }
}

File: src/main/java/com/example/hnefatafl/config/MongoConfig.kt
package com.example.hnefatafl.config

import com.mongodb.client.MongoClient
import com.mongodb.client.MongoClients
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.data.mongodb.core.MongoTemplate

@Configuration
class MongoConfig {
    @Bean
    fun mongoClient(@Value("\${spring.data.mongodb.uri}") uri: String): MongoClient {
        return MongoClients.create(uri)
    }

    @Bean
    fun mongoTemplate(
        mongoClient: MongoClient,
        @Value("\${spring.data.mongodb.database}") database: String
    ): MongoTemplate {
        return MongoTemplate(mongoClient, database)
    }
}